templates/al2023/provisioners/install-nvidia-driver.sh

```diff
@@ -29,8 +29,15 @@ echo "Installing NVIDIA ${NVIDIA_DRIVER_MAJOR_VERSION} drivers..."
 # of the AMI, we want to ensure that all three kernel modules (and also the userspace modules)
 # are on the same NVIDIA driver version. Currently, the script installs the NVIDIA GRID drivers
 # first and decides the full NVIDIA driver version that the AMI will adhere to
-EC2_GRID_DRIVER_S3_BUCKET="ec2-linux-nvidia-drivers"
-NVIDIA_DRIVER_FULL_VERSION=$(aws s3 ls --recursive s3://${EC2_GRID_DRIVER_S3_BUCKET}/ \
+if [[ "$AWS_REGION" == "us-isof-south-1" || "$AWS_REGION" == "eusc-de-east-1" || "$AWS_REGION" == "eu-isoe-west-1" ]]; then
+  EC2_GRID_DRIVER_S3_BUCKET="${BINARY_BUCKET_NAME}"
+  GRID_DRIVER_S3_SCAN_PATH="s3://${EC2_GRID_DRIVER_S3_BUCKET}/bin/nvidia-grid-drivers/"
+else
+  EC2_GRID_DRIVER_S3_BUCKET="ec2-linux-nvidia-drivers"
+  GRID_DRIVER_S3_SCAN_PATH="s3://${EC2_GRID_DRIVER_S3_BUCKET}/"
+fi
```

Comment:
can we just template the bucket?
e.g. 
```suggestion
EC2_GRID_DRIVER_S3_BUCKET=${EC2_GRID_DRIVER_S3_BUCKET:-"ec2-linux-nvidia-drivers"}
```

Reply:
The biggest reason to add `GRID_DRIVER_S3_SCAN_PATH` is because `aws s3 ls --recursive ${GRID_DRIVER_S3_SCAN_PATH}` will make ~20 mins to finish without including the `/bin/nvidia-grid-drivers/` path.